src/Microsoft.ComponentDetection.Detectors/yarn/YarnLockComponentDetector.cs

```diff
@@ -263,31 +275,56 @@ private void GetWorkspaceDependencies(IList<string> yarnWorkspaces, DirectoryInf
+                    this.ProcessWorkspaceDependency(dependencies, dependency, workspaceDependencyVsLocationMap, stream.Location);
                 }
             }
         }
     }
 
-    private void ProcessWorkspaceDependency(IDictionary<string, IDictionary<string, bool>> dependencies, KeyValuePair<string, IDictionary<string, bool>> newDependency)
+    private void ProcessWorkspaceDependency(IDictionary<string, IDictionary<string, bool>> dependencies, KeyValuePair<string, IDictionary<string, bool>> newDependency, IDictionary<string, string> workspaceDependencyVsLocationMap, string streamLocation)
     {
-        if (!dependencies.TryGetValue(newDependency.Key, out var existingDependency))
-        {
-            dependencies.Add(newDependency.Key, newDependency.Value);
-            return;
-        }
-
-        foreach (var item in newDependency.Value)
+        try
         {
-            if (existingDependency.TryGetValue(item.Key, out var wasDev))
+            if (!dependencies.TryGetValue(newDependency.Key, out var existingDependency))
             {
-                existingDependency[item.Key] = wasDev && item.Value;
+                dependencies.Add(newDependency.Key, newDependency.Value);
+                foreach (var item in newDependency.Value)
+                {
+                    // Adding 'Package.json stream's location'(in which workspacedependency of Yarn.lock file was found) as location of respective WorkSpaceDependency.
+                    this.AddLocationInfoToWorkspaceDependency(workspaceDependencyVsLocationMap, streamLocation, newDependency.Key, item.Key);
+                }
+
+                return;
             }
-            else
+
+            foreach (var item in newDependency.Value)
             {
-                existingDependency[item.Key] = item.Value;
+                if (existingDependency.TryGetValue(item.Key, out var wasDev))
+                {
+                    existingDependency[item.Key] = wasDev && item.Value;
+                }
+                else
+                {
+                    existingDependency[item.Key] = item.Value;
+                }
+
+                // Adding 'Package.json stream's location'(in which workspacedependency of Yarn.lock file was found) as location of respective WorkSpaceDependency.
+                this.AddLocationInfoToWorkspaceDependency(workspaceDependencyVsLocationMap, streamLocation, newDependency.Key, item.Key);
             }
         }
+        catch (Exception ex)
+        {
+            this.Logger.LogError(ex, "Could not process workspace dependency from file {PackageJsonStreamLocation}.", streamLocation);
+        }
+    }
+
+    private void AddLocationInfoToWorkspaceDependency(IDictionary<string, string> workspaceDependencyVsLocationMap, string streamLocation, string dependencyName, string dependencyVersion)
+    {
+        var locationMapDictionaryKey = dependencyName + "-" + dependencyVersion;
```

Comment:
Create an utility function to generate this key, and use it during insertion and querying

Reply:
Added 'GetLocationMapKey' utility in next iteration.